### PR TITLE
Fill changelog for v0.17

### PR DESCRIPTION
**Pull request by adonath**
**Description**

This PR fills the changelog for Gammapy v0.17. I've summarised the most important bug fixes, API changes and new features.

**Dear reviewer**
Pleas do a quick review @registerrier

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -1,10 +1,147 @@
...
+**What's new**
+
+Gammapy v0.17 comes with new important features, an improved sub-package
+stucture and a more uniform API.
+
+The main feature introduces in Gammapy v0.17 is event sampling. Based
+on the newly introduced `MapDatasetEventSampler` class, event lists can be
+sampled from a `MapDataset` object. The use of this class is shown in a dedicated
+`event sampling tutorial <notebooks/event_sampling.html>`__. Gammapy v0.17 now
+supports simulation and fitting of temporal models. Both is demonstrated in the
```
Comment (registerrier) [inline CHANGES.rst:22]: Both are demonstrated

File: CHANGES.rst [hunk 2]
```diff
@@ -1,10 +1,147 @@
...
+`lightcurve simulation tutorial <notebooks/light_curve_simulation.html>`__.
+A more general introduction to modelling and fitting in Gammapy is now available
```
Comment (registerrier) [inline CHANGES.rst:24]: modeling

File: CHANGES.rst [hunk 3]
```diff
@@ -1,10 +1,147 @@
...
+as a `modelling and fitting tutorial <notebooks/modeling.html>`__
```
Comment (registerrier) [inline CHANGES.rst:25]: modeling

File: CHANGES.rst [hunk 4]
```diff
@@ -1,10 +1,147 @@
...
+component using the newly introduced `SkyModel.apply_irf` and `SkyDiffuseCube.apply_irf`
+keywords. A new `GaussianTemporalModel` and `ExpDecayTemporalModel` were
+introduced.
+
+To further unify the data structures for 1D and 3D analyses a `RegionGeom`
+and `RegionNDMap` were introduced in `gammapy.maps`. These region based map classes
+are now used for the `SpectrumDataset` and `SpectrumDatasetOnOff`. The previously
+used `CountsSpectrum` class was removed.
+
+Again this release contains several API breaking changes and removal of
```
Comment (registerrier) [inline CHANGES.rst:56]: Maybe start with this part.

File: CHANGES.rst [hunk 5]
```diff
@@ -1,10 +1,147 @@
...
+The application of PSF and energy dispersion can now be configured per model
+component using the newly introduced `SkyModel.apply_irf` and `SkyDiffuseCube.apply_irf`
+keywords. A new `GaussianTemporalModel` and `ExpDecayTemporalModel` were
+introduced.
+
+To further unify the data structures for 1D and 3D analyses a `RegionGeom`
+and `RegionNDMap` were introduced in `gammapy.maps`. These region based map classes
+are now used for the `SpectrumDataset` and `SpectrumDatasetOnOff`. The previously
+used `CountsSpectrum` class was removed.
+
```
Comment (registerrier) [inline CHANGES.rst:55]: You might add:
To unify and clean the statistical calculations, counts statistic classes have been introduced to perform excess, background, significance, error, asymmetric errors and upper limits calculations. The `gammapy.stat.poisson` module has been removed. The `significance_lima` methods have been removed.

File: CHANGES.rst [hunk 6]
```diff
@@ -1,10 +1,147 @@
...
+Following the proposal in `PIG 19`_ the sub-package structure of Gammapy was
+unified. Instead of grouping the main functionality by use-case it is now
+grouped by abstractions and data levels. For this all `Dataset` classes
+were moved to the newly introduced `gammapy.datasets`. All `Maker` classes
+for data reduction from the DL3 to the DL4 data level were moved to the new
+`gammapy.makers` sub-package and all high-level `Estimator` classes were moved
+to the new `gammapy.estimators`. In addition a `gammapy.visualization` module
+was introduced to group plotting related functionality into a single namespace.
+The `gammapy.cube`, `gammapy.spectrum` and `gammapy.detect` modules were removed.
```
Comment (registerrier) [inline CHANGES.rst:35]: For the latter, the `LiMaMapEstimator` has been removed and replaced with a more general `ExcessMapEstimator` that can also compute errors and upper limit maps. The `TSMapEstimator` now uses the `MapDataset` PSF automatically.

File: CHANGES.rst (hunk 1 above)
Comment (adonath) [reply CHANGES.rst:22]: Done

File: CHANGES.rst (hunk 2 above)
Comment (adonath) [reply CHANGES.rst:24]: Done

File: CHANGES.rst (hunk 3 above)
Comment (adonath) [reply CHANGES.rst:25]: Done

File: CHANGES.rst (hunk 4 above)
Comment (adonath) [reply CHANGES.rst:56]: Done

File: CHANGES.rst (hunk 5 above)
Comment (adonath) [reply CHANGES.rst:55]: Indeed those I forgot. Fixed. 

I also added a section on the covariance handling.

File: CHANGES.rst (hunk 6 above)
Comment (adonath) [reply CHANGES.rst:35]: Done

Comment (adonath) [reply CHANGES.rst:35]: I think the `LIMaMapEstimator` was never part of a release, I'm not mentioning in...

**Comment by adonath** [comment]
Thanks @registerrier!

**Pull request merged**